pkg/imagestream/imagestream_test.go

```diff
@@ -0,0 +1,350 @@
+		} else {
+			assert.Equal(t, testImageStream.Name, builderResult.Object.Name)
+			assert.Equal(t, testImageStream.Namespace, builderResult.Object.Namespace)
+			assert.Nil(t, err)
+		}
+	}
+}
+
+func TestImageStreamGet(t *testing.T) {
+	testCases := []struct {
+		testImageStream *Builder
+		expectedError   error
+	}{
+		{
+			testImageStream: buildValidImageStreamBuilder(buildImageStreamClientWithDummyObject()),
+			expectedError:   nil,
+		},
+		{
+			testImageStream: buildInValidImageStreamBuilder(buildImageStreamClientWithDummyObject()),
+			expectedError:   fmt.Errorf("the imageStream 'name' cannot be empty"),
+		},
+		{
+			testImageStream: buildValidImageStreamBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:   fmt.Errorf("imagestreams.image.openshift.io \"network-tools\" not found"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		imageStreamObj, err := testCase.testImageStream.Get()
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, imageStreamObj.Name, testCase.testImageStream.Definition.Name)
+			assert.Equal(t, imageStreamObj.Namespace, testCase.testImageStream.Definition.Namespace)
+			assert.Nil(t, err)
+		} else {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+		}
+	}
+}
+
+func TestImageStreamExists(t *testing.T) {
+	testCases := []struct {
+		testImageStream *Builder
+		expectedStatus  bool
+	}{
+		{
+			testImageStream: buildValidImageStreamBuilder(buildImageStreamClientWithDummyObject()),
+			expectedStatus:  true,
+		},
+		{
+			testImageStream: buildInValidImageStreamBuilder(buildImageStreamClientWithDummyObject()),
+			expectedStatus:  false,
+		},
+		{
+			testImageStream: buildValidImageStreamBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedStatus:  false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		exist := testCase.testImageStream.Exists()
+		assert.Equal(t, testCase.expectedStatus, exist)
+	}
+}
+
+func TestImageStreamGetDockerImage(t *testing.T) {
+	testCases := []struct {
+		testImageStreamBuilder *Builder
+		expectedError          error
+	}{
+		{
+			testImageStreamBuilder: buildValidImageStreamBuilder(buildImageStreamClientWithDummyObject()),
+			expectedError:          nil,
+		},
+		{
+			testImageStreamBuilder: buildValidImageStreamBuilder(buildImageStreamClientNoTagsWithDummyObject()),
+			expectedError: fmt.Errorf("imageStream object network-tools in namespace " +
+				"openshift has no tags"),
+		},
+		{
+			testImageStreamBuilder: buildInValidImageStreamBuilder(buildImageStreamClientWithDummyObject()),
+			expectedError:          fmt.Errorf("the imageStream 'name' cannot be empty"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		dockerImage, err := testCase.testImageStreamBuilder.GetDockerImage()
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Nil(t, err)
+			assert.NotEqual(t, "", dockerImage)
+		} else {
+			assert.Equal(t, testCase.expectedError, err)
```

Comment:
since you're checking for the error already, could you please remove the checks from this if statement?